packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -163,17 +163,16 @@ class FlutterManifest {
     if (isModule) {
       return _flutterDescriptor['module']['androidPackage'] as String;
     }
-    if (isPlugin) {
-      final YamlMap plugin = _flutterDescriptor['plugin'] as YamlMap;
-      if (plugin.containsKey('platforms')) {
-        final YamlMap platforms = plugin['platforms'] as YamlMap;
-
-        if (platforms.containsKey('android')) {
-          return platforms['android']['package'] as String;
-        }
-      } else {
+    if (supportedPlatforms == null) {
+      // Legacy format
```

Comment:
"Legacy" is only a useful term until the new version gets deprecated and you need like "legacy 2".  Maybe "Pre-federated plugin format"?

Reply:
Makes sense! done!